src/ansys/sound/core/sound_composer/source_broadband_noise.py

```diff
@@ -0,0 +1,334 @@
+
+    def process(self, sampling_frequency: float = 44100.0):
+        """Generate the sound of the broadband noise source.
+
+        This method generates the sound of the broadband noise source, using the current broadband
+        noise data and source control.
+
+        Parameters
+        ----------
+        sampling_frequency : float, default: 44100.0
+            Sampling frequency of the generated sound in Hz.
+        """
+        if sampling_frequency <= 0.0:
+            raise PyAnsysSoundException("Sampling frequency must be strictly positive.")
+
+        if not self.is_source_control_valid():
+            raise PyAnsysSoundException(
+                "Broadband noise source control is not set. "
+                f"Use ``{__class__.__name__}.source_control``."
+            )
+
+        if self.source_bbn is None:
+            raise PyAnsysSoundException(
+                f"Broadband noise source data is not set. Use ``{__class__.__name__}.source_bbn`` "
+                f"or method ``{__class__.__name__}.load_source_bbn()``."
+            )
+
+        # Set operator inputs.
+        self.__operator_generate.connect(0, self.source_bbn)
+        self.__operator_generate.connect(1, self.source_control.control)
+        self.__operator_generate.connect(2, sampling_frequency)
+
+        # Run the operator.
+        self.__operator_generate.run()
+
+        # Get the loaded sound power level parameters.
+        self._output = self.__operator_generate.get_output(0, "field")
+
+    def get_output(self) -> Field:
+        """Get the generated sound as a DPF field.
+
+        Returns
+        -------
+        Field
+            Generated sound as a DPF field.
+        """
+        if self._output == None:
+            warnings.warn(
+                PyAnsysSoundWarning(
+                    "Output is not processed yet. "
+                    f"Use the ``{__class__.__name__}.process()`` method."
+                )
+            )
+        return self._output
+
+    def get_output_as_nparray(self) -> np.ndarray:
+        """Get the generated sound as a NumPy array.
+
+        Returns
+        -------
+        numpy.ndarray
+            Generated sound (signal samples in Pa) as a NumPy array.
+        """
+        output = self.get_output()
+
+        return np.array(output.data if output is not None else [])
+
+    def plot(self):
+        """Plot the resulting signal in a figure."""
+        if self._output == None:
+            raise PyAnsysSoundException(
+                f"Output is not processed yet. Use the '{__class__.__name__}.process()' method."
+            )
+        output = self.get_output()
+
+        time_data = output.time_freq_support.time_frequencies.data
+
+        plt.plot(time_data, output.data)
+        plt.title(output.name if len(output.name) > 0 else "Signal from broadband noise source")
+        plt.xlabel("Time (s)")
+        plt.ylabel("Amplitude (Pa)")
```

Comment:
@ansaminard here, depending if we are in narrowband or octave / 1/ octave, we will be in Pa^2/Hz (narrowband) or in Pa^2 (octave) for the display